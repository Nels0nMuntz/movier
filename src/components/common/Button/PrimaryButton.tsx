import { styled } from "@mui/material/styles";
import { FC, PropsWithChildren } from "react";
import { Typography } from "../Typography/Typography";


interface Props extends PropsWithChildren {
  icon?: React.ReactNode;
}

export const StyledButton = styled("button")(({ theme }) => ({
  position: "relative",
  display: "flex",
  alignItems: "center",
  gap: theme.spacing(1),
  paddingLeft: theme.spacing(3),
  paddingRight: theme.spacing(3),
  paddingTop: theme.spacing(1),
  paddingBottom: theme.spacing(1),
  background: theme.palette.secondary.main,
  "& *": {
    position: "relative",
    zIndex: 1,
  },
  "&:before": {
    position: "absolute",
    content: "\"\"",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    background: "#333",
    transition: "transform 0.3s ease-out",
    transform: "scaleX(0)",
    transformOrigin: "0 50%",
  },
  "&:hover": {
    background: theme.palette.common.black,
    "&:before": {
      transform: "scaleX(1)",
    }
  }
}));

export const PrimaryButton: FC<Props> = ({ icon, children }) => {
  return (
    <StyledButton>
      {icon}
      {children && (
        <Typography element="span" type="body_1">
          {children}
        </Typography>
      )}
    </StyledButton>
  )
};