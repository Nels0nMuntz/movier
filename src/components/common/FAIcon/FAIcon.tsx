import { FC } from "react";
import { IconProp } from "@fortawesome/fontawesome-svg-core"
import { FontAwesomeIcon, FontAwesomeIconProps,  } from "@fortawesome/react-fontawesome"


interface Props {
  icon: IconProp;
  color?: string;
}

export const FAIcon: FC<Props & FontAwesomeIconProps> = ({ icon, color, ...rest }) => {
  return (
    <FontAwesomeIcon 
      icon={icon} 
      color={color || "white"}
      {...rest}
    />
  )
};