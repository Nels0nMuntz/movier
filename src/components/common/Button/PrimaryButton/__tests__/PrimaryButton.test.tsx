import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { PrimaryButton } from "../PrimaryButton"
import { FAIcon } from "components/common/FAIcon/FAIcon";
import { faPlus } from "@fortawesome/free-solid-svg-icons";


describe("PrimaryButton component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<PrimaryButton />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("should render button text correctly", () => {
    const text = "Primary Button";
    render(<PrimaryButton>{text}</PrimaryButton>);
    const element = screen.getByText(text);
    expect(element).toBeInTheDocument();
  });
 
  it("should fill all avalable space", () => {
    const text = "Primary Button Fluid";
    render(<PrimaryButton fluid>{text}</PrimaryButton>);
    const element = screen.getByText(text).parentElement;
    expect(element).toHaveStyle({ "width": "100%" });
  });

  it("should render button icon correctly", () => {
    const testId = "icon-element"
    const icon = (
      <span data-testid={testId}>
        <FAIcon icon={faPlus}/>
      </span>
    )
    render(<PrimaryButton icon={icon}>Primary Button</PrimaryButton>);
    const element = screen.getByTestId(testId);
    expect(element).toBeInTheDocument();
  });
})