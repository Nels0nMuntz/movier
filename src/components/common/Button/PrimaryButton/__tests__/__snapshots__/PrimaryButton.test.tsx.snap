// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton component renders correctly 1`] = `
<button
  className="css-eos4vf"
/>
`;
